src/prezzemolo/graph.py

```diff
@@ -87,6 +88,39 @@ def breadth_first_search(self, start: Vertex[ValueType], end: Vertex[ValueType])
         # start and end are type checked inside _breadth_first_search()
         return self._breadth_first_search(start, end, vertex_2_parent=None)
 
+    def _dijkstra_search(
+        self, start: Vertex[ValueType], end: Optional[Vertex[ValueType]], vertex_2_parent: Optional[Dict[Vertex[ValueType], Optional[Vertex[ValueType]]]]
+    ) -> bool:
+        distances: Dict[Vertex[ValueType], float] = {v: float("inf") for v in self.__vertexes}
+        distances[start] = 0.0
+        priority_queue: List[Tuple[float, Vertex[ValueType]]] = [(0.0, start)]
+        predecessors: Dict[Vertex[ValueType], Optional[Vertex[ValueType]]] = {v: None for v in self.__vertexes}
+        if vertex_2_parent is not None:
+            vertex_2_parent[start] = None
+
+        while priority_queue:
+            current_distance: float
+            current_vertex: Vertex[ValueType]
+            (current_distance, current_vertex) = heappop(priority_queue)
+            if current_distance > distances[current_vertex]:
+                continue
+            for neighbor in current_vertex.neighbors:
+                new_distance = current_distance + current_vertex.get_weight(neighbor)
+                if vertex_2_parent is not None:
+                    vertex_2_parent[neighbor] = current_vertex
+                if new_distance < distances[neighbor]:
+                    distances[neighbor] = new_distance
+                    predecessors[neighbor] = current_vertex
+                    heappush(priority_queue, (new_distance, neighbor))
```

Comment:
Using a list it seems to me you may be pushing duplicate elements here. E.g. if neighbor was already in the list with a higher distance you may end up with two instances of it (which is probably not fatal but it would make the queue longer). This could be resolved by using a real [priority queue](https://docs.python.org/3/library/queue.html?highlight=priority%20queue#queue.PriorityQueue) instead of a list.